app/controllers/api/base_controller/generic.rb

```diff
@@ -193,6 +193,7 @@ def invoke_custom_action(type, resource, action, data)
 
       def invoke_custom_action_with_dialog(type, resource, action, data, custom_button)
         result = begin
+                   custom_button.invoke(resource)
```

Comment:
What other side effects does this have?

Reply:
Yeah, it queues a deliver to automate because cbs are supposed to run automate methods through the queue. That's it.

Reply:
I guess I'm trying to verify that we won't now do "the thing" twice since we're calling `custom_button.invoke` and `submit_custom_action_dialog`.  (I don't know what custom_button.invoke does)

Reply:
The only thing that ```invoke``` does at the moment is create the event, and, like you saw, queue the deliver to automate. I'm testing that that the event creation only happens once. ```submit_custom_action_dialog``` is completely different functionality.